File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/Form/Features/MinimumSizePage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.Form.Features
{
    public partial class MinimumSizePage : UserControl
    {
        public MinimumSizePage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the Click event of the objOpenDialogButton control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjOpenDialogButton_Click(object sender, EventArgs e)
        {
            //Creates and displays MinimumSizeForm instance
            MinimumSizeForm mobjDialog = new MinimumSizeForm();
            mobjDialog.MinimumSize = mobjDialog.mobjBorderPanel.Size = this.MinimumSize;
            mobjDialog.ShowDialog();
        }

        /// <summary>
        /// Handles the ValueChanged event of the objWidthNUD control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjWidthNUD_ValueChanged(object sender, EventArgs e)
        {
            //Applies Minimum Height of form
            this.MinimumSize = new Size((int)mobjWidthNUD.Value, this.MinimumSize.Height);
            //Updates panel's size according to form's MinimumSize 
            mobjBorderPanel.Size = this.MinimumSize;
        }

        /// <summary>
        /// Handles the ValueChanged event of the objHeightNUD control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjHeightNUD_ValueChanged(object sender, EventArgs e)
        {
            //Applies Minimum Width of form
            this.MinimumSize = new Size(this.MinimumSize.Width, (int)mobjHeightNUD.Value);
            //Updates panel's size according to form's MinimumSize 
            mobjBorderPanel.Size = this.MinimumSize;
        }

        /// <summary>
        /// Handles the Click event of the objBorderButton control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjBorderButton_Click(object sender, EventArgs e)
        {
            //Gets visibility state of panel
            bool IsBorderVisible = mobjBorderPanel.Visible;
            //If Border is invisible - show it 
            if (!IsBorderVisible)
            {
                mobjBorderPanel.Visible = true;
                mobjBorderButton.Text = "Hide minimum size border";
            }
            //If visible - hide it
            else
            {
                mobjBorderPanel.Visible = false;
                mobjBorderButton.Text = "Show minimum size border";
            }
        }


    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/Form/Features/MinimumSizePage.vb
Imports System.Drawing

Namespace CompanionKit.Controls.Form.Features

    Public Class MinimumSizePage

        Public Sub New()

            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()

        End Sub

        ''' <summary>
        ''' Handles the Click event of the objOpenDialogButton control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjOpenDialogButton_Click(sender As Object, e As EventArgs) Handles mobjOpenDialogButton.Click
            'Creates and displays MinimumSizeForm instance
            Dim mobjDialog As New MinimumSizeForm()
            mobjDialog.MinimumSize = Me.MinimumSize
            mobjDialog.mobjBorderPanel.Size = Me.MinimumSize
            mobjDialog.ShowDialog()
        End Sub

        ''' <summary>
        ''' Handles the ValueChanged event of the objWidthNUD control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjWidthNUD_ValueChanged(sender As Object, e As EventArgs) Handles mobjWidthNUD.ValueChanged
            'Applies Minimum Height of form
            Me.MinimumSize = New Size(CInt(mobjWidthNUD.Value), Me.MinimumSize.Height)
            'Updates panel's size according to form's MinimumSize
            mobjBorderPanel.Size = Me.MinimumSize
        End Sub

        ''' <summary>
        ''' Handles the ValueChanged event of the objHeightNUD control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjHeightNUD_ValueChanged(sender As Object, e As EventArgs) Handles mobjHeightNUD.ValueChanged
            'Applies Minimum Width of form
            Me.MinimumSize = New Size(Me.MinimumSize.Width, CInt(mobjHeightNUD.Value))
            'Updates panel's size according to form's MinimumSize
            mobjBorderPanel.Size = Me.MinimumSize
        End Sub

        ''' <summary>
        ''' Handles the Click event of the objBorderButton control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjBorderButton_Click(sender As Object, e As EventArgs) Handles mobjBorderButton.Click
            'Gets visibility state of panel
            Dim IsBorderVisible As Boolean = mobjBorderPanel.Visible
            'If Border is invisible - show it
            If Not IsBorderVisible Then
                mobjBorderPanel.Visible = True
                mobjBorderButton.Text = "Hide minimum size border"
            Else
                'If visible - hide it
                mobjBorderPanel.Visible = False
                mobjBorderButton.Text = "Show minimum size border"
            End If
        End Sub
    End Class

End Namespace